### PR TITLE
EPMRPP-91885 || Admin not assigned to a project can not open the proj…

### PR DESCRIPTION
**Pull request by BlazarQSO**
…ect, wrong project name is displayed in the url

## PR Checklist

* [x] Have you verified that the PR is pointing to the correct target branch? (`develop` for features/bugfixes, other if mentioned in the task)
* [x] Have you verified that your branch is consistent with the target branch and has no conflicts? (if not, make a rebase under the target branch)
* [x] Have you checked that everything works within the branch according to the task description and tested it locally?
* [x] Have you run the linter (`npm run lint`) prior to submission? Enable the git hook on commit in your IDE to run it and format the code automatically.
* [x] Have you run the tests locally and added/updated them if needed?
* [x] Have you checked that app can be built (`npm run build`)?
* [x] Have you made sure that all the necessary pipelines has been successfully completed?
* [x] If the task requires translations to be updated, have you done this by running the `manage:translations` script?

### CODE REVIEW

File: app/src/controllers/project/selectors.js [hunk 1]
```diff
@@ -55,7 +57,15 @@ export const projectNameSelector = (state) => projectInfoSelector(state).project
...
-  (assignedOrganizations, { organizationId }) => {
+  userAccountRoleSelector,
+  organizationsListSelector,
+  (assignedOrganizations, { organizationId }, accountRole, organizations) => {
+    const isAdmin = accountRole === ADMINISTRATOR;
+
+    if (isAdmin) {
+      return organizations.find(({ id }) => id === organizationId)?.name;
+    }
+
```
Comment (AmsterGet) [inline app/src/controllers/project/selectors.js:68]: Can we select the current org name from `activeOrganization`?

**Review by AmsterGet — CHANGES_REQUESTED** [review]
If I understand correctly, these changes will not cover the same issue for Manager (also mentioned in the ticket).
I suppose we need to change a bit our approach of defining the active project and organization.
We have several cases here:
1. Admin
In case the admin user wants to open any page within organization and/or project, just redirect him there.
2. Manager
If the user has Manager role within the org he is going to see (can be checked via assignedOrganizations), just redirect him to any project without checking it's existence.
3. Member
Need to check the assignment to org and project.

Case for non existent project/org will be handled separately as mentioned in TODO.

Please, split the checks for all described cases into logical blocks within `onBeforeRouteChange` as now it is difficult to understand what is going on there.

File: app/src/controllers/project/selectors.js (hunk 1 above)
Comment (BlazarQSO) [reply app/src/controllers/project/selectors.js:68]: Done.

File: app/src/controllers/initialData/sagas.js [hunk 2]
```diff
@@ -32,13 +32,15 @@ import {
...
 import { SET_ACTIVE_PROJECT_KEY } from 'controllers/user/constants';
+import { fetchOrganizationsAction } from 'controllers/organizations';
 import { setInitialDataReadyAction } from './actionCreators';
 import { FETCH_INITIAL_DATA } from './constants';
 
 function* fetchInitialData() {
   yield put(setTokenAction(getStorageItem(TOKEN_KEY) || DEFAULT_TOKEN));
   yield put(fetchAppInfoAction());
   yield put(fetchUserAction());
+  yield put(fetchOrganizationsAction());
```
[the remark this answers is not in the document]
Comment (BlazarQSO) [reply app/src/controllers/initialData/sagas.js:43]: Deleted

File: app/src/controllers/administrate/projects/sagas.js [hunk 3]
```diff
@@ -140,11 +140,11 @@ function* watchDeleteProject() {
 
 function* navigateToProject({ payload }) {
   const { project } = payload;
-  const { organizationSlug, projectSlug } = project;
+  const { organizationSlug, projectSlug, projectKey } = project;
 
   yield put({
     type: PROJECT_PAGE,
-    payload: { organizationSlug, projectSlug },
+    payload: { organizationSlug, projectSlug, projectKey },
```
[the remark this answers is not in the document]
Comment (BlazarQSO) [reply app/src/controllers/administrate/projects/sagas.js:147]: Deleted projectKey.

File: app/src/routes/routesMap.js [hunk 4]
```diff
@@ -319,27 +315,32 @@ const routesMap = {
   [PROJECT_PLUGIN_PAGE]: '/:projectId/plugin/:pluginPage/:pluginRoute*',
 };
 
-export const onBeforeRouteChange = (dispatch, getState, { action }) => {
+export const onBeforeRouteChange = async (dispatch, getState, { action }) => {
```
Comment (AmsterGet) [inline app/src/routes/routesMap.js:318]: Please revert it to be synchronous.

File: app/src/controllers/administrate/sagas.js [hunk 5]
```diff
@@ -37,10 +51,52 @@ function* fetchProjectData() {
...
+
+    if (projectKey) {
+      yield put(setActiveProjectKeyAction(projectKey));
+      yield put(fetchProjectAction(projectKey));
+    } else {
+      yield take(createFetchPredicate(FETCH_ORGANIZATION_BY_SLUG));
+      yield put(fetchOrganizationProjectsAction(organizationSlug));
+      yield take(createFetchPredicate(FETCH_ORGANIZATION_PROJECTS));
+
+      const organizationProjects = yield select(organizationProjectsSelector);
```
Comment (AmsterGet) [inline app/src/controllers/administrate/sagas.js:66]: Please leave a TODO comment to rewrite to it to 'fetch org project by slug' in the future.

File: app/src/controllers/administrate/sagas.js [hunk 6]
```diff
@@ -37,10 +51,52 @@ function* fetchProjectData() {
...
+      const key = organizationProjects?.items?.find(({ slug }) => slug === projectSlug)?.key;
+
+      yield put(setActiveProjectKeyAction(key));
+      yield put(fetchProjectAction(key));
+      yield put(
+        fetchDashboardsAction({
+          [SIZE_KEY]: 300,
+        }),
```
Comment (AmsterGet) [inline app/src/controllers/administrate/sagas.js:74]: Why do we need to fetch dashboards here?
I suppose it should be removed.

File: app/src/controllers/administrate/actionCreators.js [hunk 7]
```diff
@@ -14,8 +14,13 @@
...
+import { FETCH_PROJECT_DATA, FETCH_ACTIVE_PROJECT_DATA } from './constants';
 
 export const fetchProjectDataAction = () => ({
   type: FETCH_PROJECT_DATA,
 });
+
+export const fetchActiveProjectDataAction = (payload) => ({
+  type: FETCH_ACTIVE_PROJECT_DATA,
+  payload,
+});
```
Comment (AmsterGet) [inline app/src/controllers/administrate/actionCreators.js:26]: The same comment as for [this approach](https://github.com/reportportal/service-ui/pull/3876#discussion_r1648854041).
Let's not use the deprecated 'administrate' controllers and pages as they will be commissioned soon in favor of instance/organization level controllers.
Please move all your new logic to the organization controller or project controller.

**Review by AmsterGet — CHANGES_REQUESTED** [review]

File: app/src/controllers/organizations/organization/sagas.js [hunk 8]
```diff
@@ -15,32 +15,38 @@
...
+    activeOrganization = yield select(activeOrganizationSelector);
 
-    // TODO: Uncomment this line after implementation of the organizationProjects in backend
-    // yield put(fetchDataAction(PROJECTS_NAMESPACE)(URLS.organizationProjects(orgsSlug)));
+    yield put(
+      fetchDataAction(FETCH_ORGANIZATION_PROJECTS)(
+        URLS.organizationProjects(activeOrganization.id, prefParam),
+      ),
+    );
+    yield take(createFetchPredicate(FETCH_ORGANIZATION_PROJECTS));
```
[the remark this answers is not in the document]
Comment (BlazarQSO) [reply app/src/controllers/organizations/organization/sagas.js:49]: Done.

File: app/src/routes/routesMap.js (hunk 4 above)
Comment (BlazarQSO) [reply app/src/routes/routesMap.js:318]: Done.

File: app/src/controllers/administrate/sagas.js (hunk 5 above)
Comment (BlazarQSO) [reply app/src/controllers/administrate/sagas.js:66]: Done.

File: app/src/controllers/administrate/sagas.js (hunk 6 above)
Comment (BlazarQSO) [reply app/src/controllers/administrate/sagas.js:74]: We have a new projectKey, if we don’t make a request, then there will be no dashboards for this project.

File: app/src/controllers/administrate/actionCreators.js (hunk 7 above)
Comment (BlazarQSO) [reply app/src/controllers/administrate/actionCreators.js:26]: This is a repetition of the same comment in different threads, it needs to be edited in one place.

Comment (BlazarQSO) [reply app/src/controllers/administrate/actionCreators.js]: Done.

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**